mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/loader/validator/ParameterModelValidator.java

```diff
@@ -107,6 +101,40 @@ private void validateParameter(ParameterModel parameterModel, String ownerName,
 
     }
 
+    private void validateDefaultValue(ParameterModel parameterModel, String ownerName, String ownerModelType) {
+      if (parameterModel.isOverrideFromConfig() &&
+          !ownerModelType.equals(CONFIGURATION) && !ownerModelType.equals(CONNECTION_PROVIDER)) {
+        // We skip failing for configs and connection here since a different error is thrown in their own validators
+        addError(parameterModel,
+                 "Parameter '%s' in the %s '%s' is declared as a config override,"
+                     + " and must not provide a default value since one is already provided by the value"
+                     + " declared in the config parameter",
+                 parameterModel.getName(), ownerModelType, ownerName);
+      } else if (parameterModel.isRequired()) {
+        addError(parameterModel,
+                 "Parameter '%s' in the %s '%s' is required, and must not provide a default value",
+                 parameterModel.getName(), ownerModelType, ownerName);
+      }
+
+      if (!hasExpressionDefaultValue(parameterModel)) {
+        parameterModel.getType().accept(new MetadataTypeVisitor() {
```

Comment:
Since this is a new validation, we don't want apps to fail to deploy when a new runtime is used and an extension does not comply with this validation.
org.mule.runtime.module.extension.internal.loader.validation.ModelValidationUtils#isCompiletime should be used to only fail when compiling.

Reply:
disregard